packages/forklift-console-plugin/src/modules/Plans/views/details/utils/hasPipelineCompleted.tsx

```diff
@@ -0,0 +1,34 @@
+import { V1beta1PlanStatusMigrationVmsPipeline } from '@kubev2v/types';
+
+/**
+ * Check if a given migration's pipeline step has completed successfully.
+ *
+ * @param {V1beta1PlanStatusMigrationVmsPipeline} pipeline - A given migration's pipeline step
+ * @returns {boolean} - True if the migration step has completed successfully, false otherwise.
+ */
+const hasPipelineCompleted = (pipeline: V1beta1PlanStatusMigrationVmsPipeline) => {
+  if (pipeline?.error) {
+    return false;
+  }
+
+  switch (pipeline?.phase) {
+    case 'Completed':
+      return true;
+    case 'Pending':
+    case 'Running':
+    default:
+      return false;
+  }
+};
+
+/**
+ *  Check if the task isn't marked as finished, but its contained pipeline step has completed successfully.
+ *
+ * @param {string } taskPhase A given task phase
+ * @param {V1beta1PlanStatusMigrationVmsPipeline} pipeline - A given migration's pipeline step which includes the task
+ * @returns {boolean} - Returns True if the task isn't marked as finished, but its contained pipeline step has completed successfully.
+ */
+export const isTaskCompletedByItsPipeline = (
```

Comment:
```suggestion
export const isTaskCompleted = (
```

I prefer the shorter name